test/cases/azure.sh

```diff
@@ -0,0 +1,240 @@
+#!/bin/bash
+set -euo pipefail
+
+source /etc/os-release
+
+# Colorful output.
+function greenprint {
+    echo -e "\033[1;32m${1}\033[0m"
+}
+
+#TODO: Remove this once there is rhel9 support for Azure image type
+if [[ $DISTRO_CODE == rhel_90 ]]; then
+    greenprint "Skipped"
+    exit 0
+fi
+
+# Provision the software under tet.
+/usr/libexec/osbuild-composer-test/provision.sh
+
+# Apply lorax patch to work around pytoml issues in RHEL 8.x.
+# See BZ 1843704 or https://github.com/weldr/lorax/pull/1030 for more details.
+if [[ $ID == rhel ]]; then
+    sudo sed -r -i 's#toml.load\(args\[3\]\)#toml.load(open(args[3]))#' \
+        /usr/lib/python3.6/site-packages/composer/cli/compose.py
+    sudo rm -f /usr/lib/python3.6/site-packages/composer/cli/compose.pyc
+fi
+
+# We need azure-cli to talk to Azure.
+if ! hash az; then
+    # this installation method is taken from the official docs:
+    # https://docs.microsoft.com/cs-cz/cli/azure/install-azure-cli-linux?pivots=dnf
+    sudo rpm --import https://packages.microsoft.com/keys/microsoft.asc
+    echo -e "[azure-cli]
+name=Azure CLI
+baseurl=https://packages.microsoft.com/yumrepos/azure-cli
+enabled=1
+gpgcheck=1
+gpgkey=https://packages.microsoft.com/keys/microsoft.asc" | sudo tee /etc/yum.repos.d/azure-cli.repo
+
+  greenprint "Installing azure-cli"
+  sudo dnf install -y azure-cli
+  az version
+fi
+
+# We need terraform to provision the vm in azure and then destroy it
+if [ "$ID" == "rhel" ] 
+then
+    release="RHEL"
+elif [ "$ID" == "fedora" ]
+then
+    release="fedora"
+else
+    echo "Test is not running on neither Fedora nor RHEL, terminating!"
+    exit 1
+fi
```

Comment:
How about CentOS ? We offer CentOS Stream 8 as an option in the Image Builder service and that can be paired with an Azure upload target. We should be testing it as well.

Reply:
Hm, I found only `fedora` and `RHEL` here but it says `CentOS` is supported, although I'm not sure what to put in `$release`... I'll get a CentOS machine and try it.

Reply:
`RHEL`. CentOS Linux is just a rebuild. With CentOS Stream it gets a bit more complicated but the compatibility should be still there (if not, it might be a bug).